etrago/tools/execute.py

```diff
@@ -373,20 +374,74 @@ def drop_foreign_components(network):
 
         return foreign_bus, foreign_comp, foreign_series
 
+    def import_gen_from_links(network):
+        """
+        create gas generators from links in order to not lose them when
+        dropping non-electric carriers
+        """
+        gas_to_add = network.links[
+            network.links.carrier.isin(
+                [
+                    "central_gas_CHP",
+                    "OCGT",
+                    "H2_to_power",
+                    "industrial_gas_CHP",
+                ]
+            )
+        ].copy()
+        
+        # Drop generators from the links tables
+        network.links.drop(gas_to_add.index, inplace=True)
+        
+        for df in network.links_t:
+            if not network.links_t[df].empty:
+                network.links_t[df].drop(columns=gas_to_add.index,
+                                         inplace=True, errors= "ignore")
+        
+        gas_to_add.rename(columns={"bus1": "bus"}, inplace=True)
+        
+        # Discard all generators < 1kW
+        gas_to_add = gas_to_add[gas_to_add["p_nom"] >= 0.001]
+        
+        gas_to_add["Generator"] = gas_to_add["bus"] + " " + gas_to_add.index + gas_to_add["carrier"]
+        gas_to_add.set_index("Generator", drop = True, inplace= True)
+        gas_to_add = gas_to_add[gas_to_add.columns[gas_to_add.columns.isin(network.generators.columns)]]
+        
+        network.import_components_from_dataframe(gas_to_add, "Generator")
+
+        # Dealing with generators_t
+        columns_new = network.links_t.p0.columns[
+            network.links_t.p0.columns.isin(gas_to_add.index)]
+        
+        new_gen_t = network.links_t.p0[columns_new]
+        
+        network.generators_t.p = network.generators_t.p.join(new_gen_t)
+
+        return
+
     x = time.time()
     network = etrago.network
     args = etrago.args
 
     network.lines.s_nom = network.lines.s_nom_opt
-
-    # For the PF, set the P to the optimised P
+    network.links.p_nom = network.links.p_nom_opt
+    network.stores.e_nom = network.stores.e_nom_opt
+    network.storage_units.p_nom = network.storage_units.p_nom_opt
```

Comment:
I'm not sure if these lines are needed. The usage of these components is defined by a fixed p_set anyways, right?

Reply:
The mentioned lines were deleted